include/nlohmann/adl_serializer.hpp

```diff
@@ -20,14 +22,22 @@ struct adl_serializer
     @param[in] j        JSON value to read from
     @param[in,out] val  value to write to
     */
-    template<typename BasicJsonType, typename ValueType>
-    static auto from_json(BasicJsonType&& j, ValueType& val) noexcept(
+    template<typename BasicJsonType, typename U = ValueType>
```

Comment:
Do you have a better name for `U`?

Reply:
Hm, it's always going to be equal to `ValueType`, since it's never explicitly passed as a template argument. Shadowing template parameter names isn't allowed, so I can't call it `ValueType`. What about `TargetType`?

Reply:
Changed to `TargetType`.